### PR TITLE
Now blocking is more noticeable! New blocking and parrying sounds

### PR DESCRIPTION
**Pull request by DrDiasyl**
## About The Pull Request
Now whenever an attack is blocked, the sound will play and deflection with the item now plays sound too:

https://github.com/tgstation/tgstation/assets/42353186/1a0cc5b7-f2af-4d72-88d7-57cc11f5baa3

The parry.ogg was updated to a better-sounding one

https://github.com/tgstation/tgstation/assets/42353186/5ffc53d8-0b3c-4e6b-9256-b7b9735918bc

Every item now has a "block_sound" that can be set, it determines what sound is played when you block an attack with it.
Cult items no longer have their own way of fuckery to play parry sounds, now they use this system as well.

Now shield bash sound is 80% smaller after cutting out all of its noise and silence.
## Why It's Good For The Game
Adds feedback on whether the attack was blocked or not, signalling both to the attacker and defender that it didn't go through even if the animation played, which is a great QoL. Same with the deflection sound (although deflection is still pretty visible, the sound would not hurt)

New sounds add more flavour to the weapons, especially the energy sword, giving it even more badassery.

Cutting down the noise from the shield bash sound has made it better to listen and saved some kilobytes of information. And making every item use the universal system for blocking sounds, removing cultist items' own code for playing parrying sounds.
## Changelog
:cl: DrDiasyl aka DrTuxedo#0931
qol: Now blocking an attack will play a sound and display a spark effect, giving back feedback
sound: parry.ogg was updated to a better-sounding version
sound: shieldbash.ogg no longer has noise and unnecessary silence
sound: New block_shield.ogg and block_blade.ogg for shields and energy swords
code: Cultists items no longer have their own code for playing parrying sounds
/:cl:

### CODE REVIEW

**Review by Fikou — CHANGES_REQUESTED** [review]
disagree with using balloon alerts for combat notifs. it either doesnt need a visual (cause it has audio) or it should be some subtle icon/effect

**Comment by DrDiasyl** [comment]
> disagree with using balloon alerts for combat notifs. it either doesnt need a visual (cause it has audio) or it should be some subtle icon/effect

What about guns using ballon alerts when empty?
As when they're empty and you try to shoot, they show "*empty*"

**Comment by Fikou** [comment]
> > disagree with using balloon alerts for combat notifs. it either doesnt need a visual (cause it has audio) or it should be some subtle icon/effect
> 
> What about guns using ballon alerts when empty? As when they're empty and you try to shoot, they show "_empty_"

what about it? i did not say i like that it does that.

**Comment by DrDiasyl** [comment]
> > > disagree with using balloon alerts for combat notifs. it either doesnt need a visual (cause it has audio) or it should be some subtle icon/effect
> > 
> > 
> > What about guns using ballon alerts when empty? As when they're empty and you try to shoot, they show "_empty_"
> 
> what about it? i did not say i like that it does that.

Dang, you're too good. Alright, I will remove the balloon alert and will try to make an effect.
I tried to make an effect before but it didn't work, can you tell me how to make an effect?

**Comment by Fikou** [comment]
either make a temp_visual object or use one of the flick_overlay procs

**Comment by IndieanaJones** [comment]
Personally, I would do like an outline of a shield that quickly grows and fades out. Would get the idea across that the attack was blocked.

**Comment by Mothblocks** [comment]
i agree that something as core as this should be a dedicated visual effect

**Comment by DrDiasyl** [comment]
I have problems making a blocking effect.
So it should be just a white shield icon which pops up and then fades away?

**Comment by Mothblocks** [comment]
My idea was a quick yellow sparkle or something

File: code/modules/antagonists/cult/cult_bastard_sword.dm [hunk 1]
```diff
@@ -81,16 +82,10 @@
...
 		return FALSE
-	playsound(src, pick('sound/weapons/effects/ric1.ogg', 'sound/weapons/effects/ric2.ogg', 'sound/weapons/effects/ric3.ogg', 'sound/weapons/effects/ric4.ogg', 'sound/weapons/effects/ric5.ogg'), 100, 1)
 	return TRUE
 
 /obj/item/cult_bastard/hit_reaction(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
 	if(!prob(final_block_chance))
 		return FALSE
-	if(attack_type == PROJECTILE_ATTACK)
-		owner.visible_message(span_danger("[owner] deflects [attack_text] with [src]!"))
-		playsound(src, pick('sound/weapons/effects/ric1.ogg', 'sound/weapons/effects/ric2.ogg', 'sound/weapons/effects/ric3.ogg', 'sound/weapons/effects/ric4.ogg', 'sound/weapons/effects/ric5.ogg'), 100, 1)
```
Comment (JohnFulpWillard) [inline code/modules/antagonists/cult/cult_bastard_sword.dm:92]: We seem to be removing a lot of sounds here, is there a reason we want it gone?

Comment (DrDiasyl) [reply code/modules/antagonists/cult/cult_bastard_sword.dm:92]: Now reflecting plays a blocking sound of the item, so this code is redundant.
As if it stayed, reflecting would've played 2 sounds at once.

**Comment by CoiledLamb** [comment]
https://www.youtube.com/watch?v=cHeiTWZpUIU kinda reminds me of this. maybe it will inspire you?

**Comment by DrDiasyl** [comment]
> https://www.youtube.com/watch?v=cHeiTWZpUIU kinda reminds me of this. maybe it will inspire you?

What.

**Comment by MrDoomBringer** [comment]
holy shit the lightsaber deflect sound is perfect this is so cool

File: code/modules/antagonists/cult/cult_bastard_sword.dm (hunk 1 above)
Comment (DrDiasyl) [reply code/modules/antagonists/cult/cult_bastard_sword.dm:92]: Also **only** cult weapons had this kind of fuckery

File: code/game/objects/items.dm [hunk 2]
```diff
@@ -636,6 +639,7 @@
 
 	if(prob(final_block_chance))
 		owner.visible_message(span_danger("[owner] blocks [attack_text] with [src]!"))
+		playsound(src, block_sound, BLOCK_SOUND_VOLUME, TRUE)
```
Comment (Mothblocks) [inline code/game/objects/items.dm:642]:
```suggestion
		playsound(src, block_sound, BLOCK_SOUND_VOLUME, vary = TRUE)
```

**Review by Mothblocks — CHANGES_REQUESTED** [review]
Please document where your new sounds are from

**Comment by DrDiasyl** [comment]
> Please document where your new sounds are from 

Oh no...
Well, the shield blocking sound is made by me using shield_bash.ogg we already had in game.

I made the sword parrying sound combining a sound from free sounds website, and also a genhit.ogg we already had in the game.

But for the energy sword sound I took it from this YouTube video, I guess I should credit it then https://youtu.be/mGleORm5rXM

**Comment by Mothblocks** [comment]
You cannot use that sound then, we don't own the copyrights to it

Yes I know the existing energy swords are too but we really should be migrating those off at some point, we shouldn't be adding any more

**Comment by DrDiasyl** [comment]
> You cannot use that sound then, we don't own the copyrights to it
> 
> Yes I know the existing energy swords are too but we really should be migrating those off at some point, we shouldn't be adding any more
> 
> From that own video:
> 
> > This sound is most likely not produced by myself, but rather taken from somewhere else. That doesn't mean that I stole the sound. (???? <---)

Well shit, I will have to make my own sound then.

**Comment by DrDiasyl** [comment]
I'm sorry @MrDoomBringer but that sound will be no more

**Comment by Mothblocks** [comment]
need a picture of what it looks like now if it's not the balloon alert anymore

**Comment by Mothblocks** [comment]
oh wait if it's only sounds then that sucks. i am hard of hearing. you need some sort of visual effect

**Comment by DrDiasyl** [comment]
> oh wait if it's only sounds then that sucks. i am hard of hearing. you need some sort of visual effect. talk to the spriters to get you one, it doesn't have to be very advanced (just not a balloon alert), i was thinking something like a quick spark or something

🪦 ok

**Comment by DrDiasyl** [comment]
https://github.com/tgstation/tgstation/assets/42353186/1d3df84e-02eb-43ca-ab80-0245d6f4c150

I made two different effects, one is a flash and the other one is a shield icon.
Which do you like most?

**Comment by DrDiasyl** [comment]
Updated videos

**Comment by DrDiasyl** [comment]
Also I have removed old saber block sound, now it's fully made out of existing sounds in our codebase

**Pull request merged**